File: backend/hotelEase/hotelEase.Model/RoomAvailability.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace hotelEase.Model
{
    public class RoomAvailability
    {
        public int Id { get; set; }

        public int RoomId { get; set; }

        public DateTime Date { get; set; }

        public int Status { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedTime { get; set; }
    }
}

File: UI/hotelease_desktop/lib/models/room_availability.dart
class RoomAvailability {
  int? roomId;
  DateTime? date;
  int? status; // 0 = available, 1 = booked, 2 = limited

  RoomAvailability({this.roomId, this.date, this.status});

  factory RoomAvailability.fromJson(Map<String, dynamic> json) {
    return RoomAvailability(
      roomId: json['roomId'] as int?,
      date: DateTime.parse(json['date']),
      status: json['status'] as int?,
    );
  }
}
